paddle/fluid/operators/parallel_do_op.cc

```diff
@@ -143,11 +143,17 @@ class ParallelDoOp : public framework::OperatorBase {
       PADDLE_ENFORCE(scope.FindVar(param)->IsType<LoDTensor>(),
                      "Only support parameter type as LoDTensor");
       auto &src = scope.FindVar(param)->Get<LoDTensor>();
-      for (size_t i = 0; i < sub_scopes.size(); ++i) {
-        auto &place = places[i];
-        auto *sub_scope = sub_scopes[i];
-        auto *dst = sub_scope->Var(param)->GetMutable<LoDTensor>();
-        framework::TensorCopy(src, place, dst);
+
+      // make a copy of each GPU place, starting from the second
+      // place
+      for (size_t i = 1; i < sub_scopes.size(); ++i) {
+	auto &place = places[i];
+	auto *sub_scope = sub_scopes[i];
+	auto exists = sub_scope->FindVarLocally(param) != nullptr;
+	if (!exists && platform::is_gpu_place(place)) {
+          auto *dst = sub_scope->Var(param)->GetMutable<LoDTensor>();
+          framework::TensorCopy(src, place, dst);
+	}
```

Comment:
parallel_do_op.cc:150:  Tab found; better to use spaces  [whitespace/tab] [1]
parallel_do_op.cc:151:  Tab found; better to use spaces  [whitespace/tab] [1]
parallel_do_op.cc:152:  Tab found; better to use spaces  [whitespace/tab] [1]
parallel_do_op.cc:153:  Tab found; better to use spaces  [whitespace/tab] [1]
parallel_do_op.cc:156:  Tab found; better to use spaces  [whitespace/tab] [1]